rules/aip0122/http_url_pattern.go

```diff
@@ -23,19 +23,56 @@ import (
 )
 
 // HTTP URL pattern shouldn't include underscore("_")
-var camelCase = &lint.MethodRule{
-	Name: lint.NewRuleName("core", "0122", "http-url-pattern"),
+var httpURICase = &lint.MethodRule{
+	Name: lint.NewRuleName("core", "0122", "http-uri-case"),
 	URI:  "https://aip.dev/122#guidance",
-	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
+	LintMethod: func(m *desc.MethodDescriptor) (problems []lint.Problem) {
 		// Establish that the URI does not include a `_` character.
 		for _, httpRule := range utils.GetHTTPRules(m) {
-			if strings.Contains(httpRule.URI, "_") {
-				return []lint.Problem{{
-					Message:    "HTTP URL should use camel case, but not snake case.",
+			parsedURI := parseURI(httpRule.URI)
+			if strings.Contains(parsedURI.pattern, "_") {
+				problems = append(problems, lint.Problem{
+					Message:    "HTTP URI patterns should use camel case, not snake case.",
 					Descriptor: m,
-				}}
+				})
+			}
+			for _, v := range parsedURI.vars {
```

Comment:
Shouldn't we create a separate rule for checking variable names?

Reply:
It is fundamentally the same string, just different pieces of it. I could be convinced either way.

Reply:
I _think_ I very slightly prefer a single rule. These really are part and parcel the same thing.

Reply:
Okay, thanks!